src/shaka_scheme/system/gc/GCList.hpp

```diff
@@ -5,25 +5,31 @@
 #ifndef SHAKA_SCHEME_GCLIST_HPP
 #define SHAKA_SCHEME_GCLIST_HPP
 
-#include "shaka_scheme/system/gc/GCData.hpp"
-
 namespace shaka {
 
-    class GCList {
+    namespace gc {
+
+        class GCData;
 
-    public:
-        GCList();
-        ~GCList();
+        class GCList {
+        public:
 
-        bool is_empty();
-        int get_size();
-        void add_data(GCData *data);
-        void sweep();
+            GCList();
+            ~GCList();
+            GCList(const GCList& other) = delete;
+            GCList(GCList&& other);
 
-    private:
-        GCData *head;
-        int size;
+            bool is_empty();
+            int get_size();
```

Comment:
`size` is too ambiguous. Do you mean number of allocated objects? Or did you mean the total amount of memory currently being consumed by the garbage collector?

Reply:
`size` refers to the number of allocated GCData objects in GCList. I will make the private member less ambiguous.